local-volume/provisioner/pkg/deleter/deleter.go

```diff
@@ -126,69 +126,53 @@ func (d *Deleter) asyncDeleteFilePV(pv *v1.PersistentVolume, mountPath string, c
 		d.RuntimeConfig.Recorder.Eventf(pv, v1.EventTypeWarning, common.EventVolumeFailedDelete,
 			deletingLocalPVErr.Error())
 		glog.Error(deletingLocalPVErr)
-		return err
+		return
 	}
 
 	glog.Infof("Deleted PV %q", pv.Name)
 
+}
+
+func (d *Deleter) deleteFilePV(pv *v1.PersistentVolume, mountPath string, config common.MountConfig) error {
+	glog.Infof("Deleting PV file volume %q contents at hostpath %q, mountpath %q", pv.Name, pv.Spec.Local.Path,
+		mountPath)
+	if err := d.VolUtil.DeleteContents(mountPath); err != nil {
+		return err
+	}
+
 	return nil
 }
 
-// cleanupBlockPV is meant to execute cleanup concurrently with other such cleanup processes
-// cmdio is a param that can suppoly io to the script being invoked. Its main purpose currently is
-// to enable better testing of the code path with output validation etc.
-func (d *Deleter) cleanupBlockPV(pv *v1.PersistentVolume, mountPath string, config common.MountConfig) error {
+func (d *Deleter) cleanupBlockPV(pv *v1.PersistentVolume, blkdevPath string, config common.MountConfig) error {
+
 	// Make absolutely sure here that we are not deleting anything outside of mounted dir
-	if !strings.HasPrefix(mountPath, config.MountDir) {
-		err := fmt.Errorf("Unexpected error pv %q mountPath %s but mount dir is %s", pv.Name, mountPath,
+	if !strings.HasPrefix(blkdevPath, config.MountDir) {
```

Comment:
Can we have this check for filesystem type too?

Reply:
done